### ISSUE TITLE
Reverse pagination: hasPreviousPage and before:

### ISSUE THREAD

**Issue by simonw** [simonw/datasette-graphql#17]
For the non-alpha release I'm going to support forwards pagination only - `after:` and `endCursor` and `hasNextPage`. I can support `startCursor` too at this point even though it's not useful for anything yet.

> I'll split `hasPreviousPage` and `before:` into a separate ticket.
_Originally posted by @simonw in https://github.com/simonw/datasette-graphql/issues/2#issuecomment-668782934_

**Comment by simonw** [comment]
This is likely blocked by Datasette core support for reverse pagination: https://github.com/simonw/datasette/issues/916

**Comment by simonw** [comment]
Worth noting that edge order has some rules in https://relay.dev/graphql/connections.htm#sec-Edge-order

> You may order the edges however your business logic dictates, and may determine the ordering based upon additional arguments not covered by this specification. But the ordering must be consistent from page to page, and importantly, *The ordering of edges should be the same when using `first`/`after` as when using `last`/`before`, all other arguments being equal.* It should not be reversed when using `last`/`before`. More formally:
> 
> -   When `before: cursor` is used, the edge closest to `cursor` must come **last** in the result `edges`.
> -   When `after: cursor` is used, the edge closest to `cursor` must come **first** in the result `edges`.

**Comment by simonw** [comment]
This is pretty low priority since `sort_desc:` can be used with `first:` and `after:` at the moment.
